hypertrace-gradle-docker-plugin/src/main/java/org/hypertrace/gradle/docker/DockerPlugin.java

```diff
@@ -113,6 +113,9 @@ private String getBranchTag() {
       // Use the value of CIRCLE_BRANCH/ GITHUB_REF environment variable if defined (that is, the branch name used
       // to build in CI), otherwise for local builds use 'test'
       return getEnvironmentVariable("CIRCLE_BRANCH")
+          // GITHUB_HEAD_REF Only set for pull request events. The name of the head branch.
```

Comment:
The logic is fine (pending the test that JC asked for), but I'm curious as to why we need it? This is used for setting a version for per-merge docker images, so should be irrelevant for pull requests - we never publish images from a pull request.

Reply:
this was for e2e tests as we do build on PRs in case of e2e tests and use those images in `docker-compose`. Earlier we used to produce images with `test` tag but now that we have added `GITHUB_REF` it produces images with tag as Ravi mentioned above. 

This will help us to get image with branch name as tag on PR and we can use it in e2e.

Reply:
Unless I'm misunderstanding, changing this logic doesn't actually solve that problem. We're making this smarter to get the branch name in more scenarios, but that means on the running side, we have to handle those same scenarios in the same way to provide the expected tag.

We have a number of alternatives, none of which would require a change here :
- Use the tag `${GITHUB_REF:-test}` and leave the plugin logic as is - if ref is available, both the plugin and the expression would pick it up, else both would default to the local build tag `test, same as in the plugin prior to the changes today.
- Change the logic when building the image for testing such that it always gets a `latest` tag, then hardcode that (I think `-Prelease` flag would accomplish this)
- Change the workflows so the tests never run on push, only on PR.

My desire is to get the CI knowledge out of this plugin, rather than adding more into it - especially if doing so int

Reply:
I managed to get it resolved by running e2e test in `pull_request_target` so images will be tagged with main tag while building. so I am fine with skipping this change and if we need we can go with `${GITHUB_REF:-test}` as you suggested.

Reply:
I feel second option will be better. The way to pass tag via CLI is something that will solve this issue and will also provide us flexibility.